src/main/java/net/neoforged/neoforge/common/MonsterRoomHooks.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.common;
+
+import java.util.ArrayList;
+import java.util.Objects;
+import net.minecraft.core.registries.Registries;
+import net.minecraft.util.RandomSource;
+import net.minecraft.util.random.Weight;
+import net.minecraft.util.random.WeightedEntry;
+import net.minecraft.util.random.WeightedRandom;
+import net.minecraft.world.entity.EntityType;
+import net.neoforged.bus.api.SubscribeEvent;
+import net.neoforged.fml.common.EventBusSubscriber;
+import net.neoforged.neoforge.internal.versions.neoforge.NeoForgeVersion;
+import net.neoforged.neoforge.registries.datamaps.DataMapsUpdatedEvent;
+import net.neoforged.neoforge.registries.datamaps.builtin.NeoForgeDataMaps;
+
+@EventBusSubscriber(bus = EventBusSubscriber.Bus.GAME, modid = NeoForgeVersion.MOD_ID)
+public class MonsterRoomHooks {
+    private static final ArrayList<MobEntry> monsterRoomMobs = new ArrayList<>();
```

Comment:
I would make this mutable and replace the list to avoid any possible concurrent modifications.